app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt

```diff
@@ -110,7 +111,7 @@ class EditProfileFragment : Fragment() {
             .observe(this, Observer {
                 Snackbar.make(rootView.editProfileCoordinatorLayout, it, Snackbar.LENGTH_LONG).show()
                 if (it == USER_UPDATED) {
-                    activity?.onBackPressed()
+                    activity?.let { act -> if (act is MainActivity) act.onSuperBackPressed() }
                 }
```

Comment:
@liveHarshit  I have done it this way instead of using `if(activity is MainActivity) activity.onSuperBackPressed()` because the following error occurred for this method

`Smart cast to 'MainActivity' is impossible, because 'activity' is a property that has open or custom getter`

Reply:
Then use a value - `val thisActivity = activity` then cast for thisActivity.

Reply:
okay! I tried that also but thought this was cleaner. will implement this